addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -86,7 +106,20 @@ export function hasIndexFile(dirPath: string): Promise<Boolean> {
  */
 export function getDependentFiles(fileName: string, rootPath: string): Promise<ModuleMap> {
   const globSearch = denodeify(glob);
-  return globSearch(path.join(rootPath, '**/*.*.ts'), { nodir: true })
+  return globSearch(path.join(rootPath, '**/*.ts'), { nodir: true })
+    .then((files: string[]) => {
+      // Filter out the spec file associated with to-be-promoted component unit.
+      // Also filter out index files.
+      let fileBaseName = path.basename(fileName, '.ts');
+      return files
+        .filter((file) => {
+          if (path.extname(path.basename(file, '.ts')) === '.spec') {
```

Comment:
I think that `getDependentFiles` shouldn't lie to you. Return the spec here as well, and filter it out when you're actually using it.